### PR TITLE
Create allergies exercise

### PR DESCRIPTION
**Pull request by sshine**
I'm pretty happy to have figured out the `QCheck` library.

I'm not sure if I followed all the right procedures for creating new exercise instances.

### CODE REVIEW

**Comment by sshine** [comment]
I've created a negative test and made `make` run nicely formatted tests:

    $ make
    dune clean
    dune runtest
            test alias runtest
    random seed: 493480375
    generated error fail pass / total     time test name
    [✓] 1000    0    0 1000 / 1000     0.0s prop_allergic_to_single_allergens
    [✓] 1000    0    0 1000 / 1000     0.0s prop_allergic_to_negative_single_allergens
    [✓] 1000    0    0 1000 / 1000     0.0s prop_allergic_to_multiple_allergens
    [✓] 1000    0    0 1000 / 1000     0.0s prop_allergies_single_allergens
    [✓] 1000    0    0 1000 / 1000     0.0s prop_allergies_multiple_allergens
    ================================================================================
    success (ran 5 tests)

- [ ] Add support for shrinking.
- [ ] Customize pretty-printer so that it prints "expected" and "actual".

File: exercises/allergies/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+allergens that score 256, 512, 1024, etc.).  Your program should
+ignore those components of the score.  For example, if the allergy
+score is 257, your program should only report the eggs (1) allergy.
+
+## Getting Started
+For installation and learning resources, refer to the
+[exercism help page](http://exercism.io/languages/ocaml).
+
+## Installation
+To work on the exercises, you will need `Opam` and `Core`. Consult [opam](https://opam.ocaml.org) website for instructions on how to install `opam` for your OS. Once `opam` is installed open a terminal window and run the following command to install core:
```
Comment (stevejb71) [inline exercises/allergies/README.md]: Looks like you only need Base for this exercise, which is a lighter dependency than Core.

File: .travis-ci.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Edit this for your own project dependencies
-OPAM_DEPENDS="ocamlfind core ounit react ppx_deriving"
+OPAM_DEPENDS="ocamlfind core ounit qcheck react ppx_deriving"
```
Comment (stevejb71) [inline .travis-ci.sh]: Change core to base.

File: exercises/allergies/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+```bash
+opam install core
+```
+
+To run the tests you will need `QCheck`. Install it using `opam`:
```
Comment (stevejb71) [inline exercises/allergies/README.md]: We've not used qcheck before. I think it's great you've introduced it, but a one line intro here might be helpful.

File: .travis-ci.sh (hunk 2 above)
Comment (sshine) [reply .travis-ci.sh]: .travis-ci.sh is not specific to this exercise.

File: exercises/allergies/README.md (hunk 1 above)
Comment (sshine) [reply exercises/allergies/README.md]: You're right. Can one install base without core? I copy-pasted the README. I'll change this here, but perhaps it needs to be changed throughout the exercises.

File: exercises/allergies/README.md (hunk 3 above)
Comment (sshine) [reply exercises/allergies/README.md]: Good idea. I'll update that.

File: exercises/allergies/README.md (hunk 1 above)
Comment (stevejb71) [reply exercises/allergies/README.md]: You can install base without core. It should be changed throughout the exercises, but on a separate PR.

File: .travis-ci.sh (hunk 2 above)
Comment (stevejb71) [reply .travis-ci.sh]: Let's leave it then.

File: exercises/allergies/README.md (hunk 1 above)
Comment (sshine) [reply exercises/allergies/README.md]: I think I'd prefer if we swept over all README.md's at once to fix Core -> Base.

**Pull request merged**